network-protection/network-protection-internal/src/main/java/com/duckduckgo/networkprotection/internal/network/NetPInternalEnvDataStore.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.internal.network
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import javax.inject.Inject
+
+class NetPInternalEnvDataStore @Inject constructor(
+    context: Context,
+) {
+
+    private val preferences: SharedPreferences by lazy { context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE) }
+    var netpCustomEnvironmentUrl: String?
+        get() = preferences.getString(KEY_NETP_ENVIRONMENT_URL, null)
+        set(value) = preferences.edit { putString(KEY_NETP_ENVIRONMENT_URL, value) }
+    var useNetpCustomEnvironmentUrl: Boolean
+        get() = preferences.getBoolean(KEY_NETP_USE_ENVIRONMENT_URL, false)
+        set(enabled) = preferences.edit { putBoolean(KEY_NETP_USE_ENVIRONMENT_URL, enabled) }
+
+    companion object {
+        private const val FILENAME = "com.duckduckgo.netp.internal.env.store.v1"
```

Comment:
Do we really need a separate file for this? I see that we have a separate shared prefs for MTU (`NetPInternalMtuProvider`) and system apps exclusion (`NetPInternalExclusionListProvider`)

Reply:
ouch, I missed this question I was too quick merging. Sorry about that.

Yeah, probably we didn't need a separate shared prefs. Will take the opportunity of a next PR (whatever that may be) and fix this.

Sorry again!